flow/layers/clip_rect_layer.cc

```diff
@@ -4,38 +4,41 @@
 
 #include "flutter/flow/layers/clip_rect_layer.h"
 
-#if defined(OS_FUCHSIA)
-#include "apps/mozart/lib/skia/type_converters.h" // nogncheck
-#include "apps/mozart/services/composition/nodes.fidl.h" // nogncheck
-#endif  // defined(OS_FUCHSIA)
-
 namespace flow {
 
-ClipRectLayer::ClipRectLayer() {}
+ClipRectLayer::ClipRectLayer() = default;
 
-ClipRectLayer::~ClipRectLayer() {}
+ClipRectLayer::~ClipRectLayer() = default;
 
 void ClipRectLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
-  PrerollChildren(context, matrix);
-  if (!context->child_paint_bounds.intersect(clip_rect_))
-    context->child_paint_bounds.setEmpty();
-  set_paint_bounds(context->child_paint_bounds);
+  SkRect child_paint_bounds = SkRect::MakeEmpty();
+  PrerollChildren(context, matrix, &child_paint_bounds);
+
+  if (child_paint_bounds.intersect(clip_rect_)) {
+    set_paint_bounds(child_paint_bounds);
+  }
 }
 
 #if defined(OS_FUCHSIA)
 
-void ClipRectLayer::UpdateScene(SceneUpdateContext& context,
-                                mozart::Node* container) {
-  auto node = mozart::Node::New();
-  node->content_clip = mozart::RectF::From(clip_rect_);
-  UpdateSceneChildrenInsideNode(context, container, std::move(node));
+void ClipRectLayer::UpdateScene(SceneUpdateContext& context) {
+  FTL_DCHECK(needs_system_composite());
+
+  // TODO(MZ-138): Need to be able to specify an origin.
```

Comment:
I think the transform is sufficient to set the origin (and we already apply it).  So let's remove this TODO here.